src/unix/fs.c

```diff
@@ -1566,11 +1693,16 @@ int uv_fs_copyfile(uv_loop_t* loop,
                    const char* new_path,
                    int flags,
                    uv_fs_cb cb) {
+  uv_work_t* work;
+  uv_work_options_t options;
+
   INIT(COPYFILE);
 
   if (flags & ~(UV_FS_COPYFILE_EXCL |
                 UV_FS_COPYFILE_FICLONE |
                 UV_FS_COPYFILE_FICLONE_FORCE)) {
+    if (work)
+      uv__free(work);
     return UV_EINVAL;
   }
 
```

Comment:
Introducing a memory allocation for each fs request is kind of regrettable.  Do you think there is a way to avoid that?

Reply:
The FS (and DNS) code formerly took an internal route to the threadpool using the hidden `uv__work` member hidden in those requests. Now they must formally submit work to the executor like everyone else.

Thus they need a `uv_work_t`.

One solution to avoid per-request allocation is to pre-allocate some `uv_work_t`'s, statically or during initialization, and maintain a list of free `uv_work_t`'s. If the pre-allocated set runs out then we can allocate new `uv_work_t`'s and either add them to the free list when they are done (up to some high water mark) or free them afterwards.